File: app/src/am_hoanghh/issues10/ViewPager/MyPagerAdapter.java
package issues10.ViewPager;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import issues10.All.AllFragment;
import issues10.Favorite.FavoriteFragment;

public class MyPagerAdapter extends FragmentStateAdapter {
    private final Fragment[] mFragments;

    public MyPagerAdapter(FragmentActivity fa) {
        super(fa);
        mFragments = new Fragment []{
            new AllFragment(),
            new FavoriteFragment()
        };
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        return mFragments[position];
    }

    @Override
    public int getItemCount() {
        return mFragments.length;
    }

    public Fragment getFragment(int position) {
        return mFragments[position];
    }
}

File: app/src/am_hoanghh/issues13/ex2/viewpager/MyPagerAdapter.kt
package issues13.ex2.viewpager

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import issues13.ex2.view.AllFragment
import issues13.ex2.view.FavoriteFragment

class MyPagerAdapter(fragmentActivity: FragmentActivity) :
    FragmentStateAdapter(fragmentActivity) {

    override fun getItemCount(): Int = 2

    override fun createFragment(position: Int): Fragment {
        return when (position) {
            1 -> FavoriteFragment()
            else -> AllFragment()
        }
    }
}
